mita/openstack.py

```diff
@@ -376,14 +359,16 @@ def get_private_ip(self):
         private_ip = node_detail.private_ips[0].encode("ascii", "ignore")
         return private_ip
 
-    def get_volume(self, name):
-        """ Return libcloud.compute.base.StorageVolume """
-        driver = self.driver
-        volumes = driver.list_volumes()
-        try:
-            return [v for v in volumes if v.name == name][0]
-        except IndexError:
-            raise RuntimeError("Unable to get volume")
+    def get_volume(self, name: str) -> StorageVolume:
+        """Retrieve the StorageVolume instance using the given name."""
+        url = f"/volumes?name={name}"
+        object_ = self.driver_v2.volumev2_connection.request(url).object
+
+        if len(object_.get("volumes", [])) != 1:
```

Comment:
The old code did not care about multiple volumes with the same name, but here we're checking that `len()` is exactly one. Let's rewrite this to `if not volumes:`

Reply:
changed.